src/constructor.ts

```diff
@@ -118,10 +111,31 @@ function makeDomainFunction<
               : formatSchemaErrors(envResult.error.issues),
           })
         }
-        return handler(result.data, envResult.data)
+        return handler(result.data as I, envResult.data as E)
       })
     } as DomainFunction<Awaited<Output>>
   }
 }
 
+// deno-lint-ignore ban-types
+const objectSchema: ParserSchema<{}> = {
+  // deno-lint-ignore require-await
+  safeParseAsync: async (data: unknown) => {
+    if (data == null || typeof data !== 'object') {
```

Comment:
`typeof [] === "object"` though..
This is the only safe way I know to assure that:
```ts
Object.prototype.toString.call({}) === "[object Object]"
Object.prototype.toString.call([]) === "[object Array]"
Object.prototype.toString.call(null) === "[object Null]"
```

Check out [this function](https://github.com/gustavoguichard/make-service/blob/main/src/internals.ts#L31)

Reply:
Yeah, I know the type was wrong (it is in the todo to review these types). But now I see the runtime check is wrong as well!